core/payment-driver/zksync/src/driver.rs

```diff
@@ -329,7 +342,28 @@ Mind that to be eligible you have to run your app at least once on testnet -
             .into_iter()
             .map(|allocation| allocation.remaining_amount)
             .sum();
-        Ok(msg.amount <= (account_balance - total_allocated_amount))
+
+        // NOTE: `wallet::get_tx_fee` accepts an _recipient_ address which is unknown at the moment
+        // so the _sender_ address is provider. This might bias fee calculation, because transaction
+        // to new account is little more expensive.
+        let tx_fee_cost = wallet::get_tx_fee(&msg.address, network).await?;
+        let total_txs_cost = BigDecimal::from(*ALLOCATE_FOR_N_TRANSACTIONS) * tx_fee_cost;
+        let max_allocation_surcharge =
```

Comment:
That's not `max` allocation surcharge. Max is the constant.
```suggestion
        let allocation_surcharge =
```